### ISSUE TITLE
Raise error when ion not in database

### ISSUE THREAD

**Issue by wtbarnes** [wtbarnes/fiasco#28]
Currently, you can create ions like,

```python
>>> import fiasco; import astropy.units as u
>>> foo = fiasco.Ion('H 500', [1e6]*u.K)
```

and no error will be raised (until of course you try to access data). Obviously this is nonsense since there is no ionization state past `H II` (or `H +1`), fully ionized hydrogen.

A few questions:

* Should we check for `stage <= Z+1` and raise an error if not satisfied?
* Should we check the database and see if the ion exists and if not raise an error
* Should we leverage Plasmapy more heavily to judge the "validity" of the requested ion?

Once this is sorted, could remove some of the checking in `Element` that is done when creating ion instances.

**Comment by kdere** [comment]
there used to be a check to see whether the ion was in masterlist.ions but when I was working on new ions, this got  in the way and I removed the check.  I should probably put the check back in with a way to ignore it.

**Comment by wtbarnes** [comment]
There's now a check in `IonBase` to make sure the ion is in the database. Otherwise, an error is raised. Adding this check incurred an unfortunate performance hit for large collections of ions. This is fixed by #32
